### PR TITLE
Ignore suppressed issues in sarif reports

### PR DESCRIPTION
**Pull request by The-Jonsey**
Ignore suppressions in SARIF reports, as them being suppressed should exclude them from the violations, and related quality gates in the warnings ng plugin 

### Testing done

Unit tests were updated and was run against a jenkins instance passing in a sarif file with suppressions, which were ignored correctly

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by The-Jonsey** [comment]
The check failures all seem to be either unrelated test timeouts or dependency-check timing out repeatedly, not sure if there's anything that needs doing about that

File: src/main/java/edu/hm/hafner/analysis/parser/violations/SarifAdapter.java [hunk 1]
```diff
@@ -40,6 +40,12 @@ void extractAdditionalProperties(final IssueBuilder builder, final Violation vio
         }
     }
 
+    @Override
+    boolean isValid(final Violation violation) {
+        return !(violation.getSpecifics().containsKey("suppressed")
+            && violation.getSpecifics().get("suppressed").equals("true"));
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/violations/SarifAdapter.java:46]: SImplify with `getOrDefault("suppressed", "false")`

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
> The check failures all seem to be either unrelated test timeouts or dependency-check timing out repeatedly, not sure if there's anything that needs doing about that

Thanks for noting, I forgot to remove the invocation from the pom.xml. This part of the build is now configured in a profile.

File: src/main/java/edu/hm/hafner/analysis/parser/violations/SarifAdapter.java (hunk 1 above)
Comment (The-Jonsey) [reply src/main/java/edu/hm/hafner/analysis/parser/violations/SarifAdapter.java]: Fair point! updated to match

**Comment by uhafner** [comment]
I think I fixed the timeout problems now, can you please merge with main?

**Pull request merged**